contracts/ColonyTask.sol

```diff
@@ -137,34 +137,50 @@ contract ColonyTask is ColonyStorage, DSMath {
     return taskCount;
   }
 
-  function proposeTaskChange(bytes _data, uint256 _value, uint8 _role) public returns (uint256 transactionId) {
-    var (sig, taskId) = deconstructCall(_data);
-
-    Task storage task = tasks[taskId];
-    require(task.roles[_role].user == msg.sender);
-    require(!task.finalized);
-
-    uint8[2] storage _reviewers = reviewers[sig];
-    require(_reviewers[0] != 0 || _reviewers[1] != 0);
-    require(_reviewers[0] == _role || _reviewers[1] == _role);
-
-    transactionId = IColony(this).submitTransaction(_data, _value, _role);
+  function getTaskChangeNonce() public view returns (uint256) {
+    return taskChangeNonce;
   }
 
-  function approveTaskChange(uint256 _transactionId, uint8 _role) public {
-    Transaction storage _transaction = transactions[_transactionId];
-    bytes memory _data = _transaction.data;
-    var (sig, taskId) = deconstructCall(_data);
+  function executeTaskChange(
+    uint8[] _sigV,
+    bytes32[] _sigR,
+    bytes32[] _sigS,
+    uint256 _value,
+    bytes _data) public 
+  {
+    // Allow for 2 reviewers
+    require(_sigR.length == 2);
+    require(_sigR.length == _sigS.length && _sigR.length == _sigV.length);
 
+    var (sig, taskId) = deconstructCall(_data);
     Task storage task = tasks[taskId];
-    require(task.roles[_role].user == msg.sender);
     require(!task.finalized);
-
+    
     uint8[2] storage _reviewers = reviewers[sig];
-    require(_reviewers[0] != 0 || _reviewers[1] != 0);
-    require(_reviewers[0] == _role || _reviewers[1] == _role);
+    uint8 r1 = _reviewers[0];
+    uint8 r2 = _reviewers[1];
+    require(r1 != 0 || r2 != 0);
```

Comment:
Took me a while to realise that this is `||` and not `&&` because `0` represents the manager. Can we add a comment noting that?

Reply:
I think we should also comment this is the line stopping someone calling an arbitrary function on the contract.

Reply:
Also that this line is the only thing stopping someone calling an arbitrary function on the contract.